frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -624,11 +633,14 @@ export default ({
       this.ephemeral.infiniteLoading = $state
       if (this.ephemeral.messagesInitiated === undefined) {
         // NOTE: this infinite handler is being called once which should be ignored
-        // before calling the setInitMessages function
+        //       before calling the setInitMessages function
+        return
+      } else if (this.currentChatRoomId !== this.renderingChatRoomId) {
+        // NOTE: should render messages after the current chatroom state is initiated
         return
       }
       this.renderMoreMessages(!this.ephemeral.messagesInitiated).then(completed => {
```

Comment:
See comment above about `this.renderMoreMessages`. Also, this most probably needs a `.catch` part2

Reply:
Agreed, if there's an exception it should be logged with `console.error` in a `.catch`

Reply:
Nice catch.